src/util.ts

```diff
@@ -130,3 +130,48 @@ export function joinAll(a: ArrayBuffer[]): ArrayBuffer {
 
     return buffer;
 }
+
+// Consumes data:
+//   PrivateToken challenge="abc...", token-key="123...",
+//   PrivateToken challenge="def...", token-key="234...",
+//   PrivateToken challenge=ghi..., token-key=345...
+// Parse WWW-Authenticate according to RFC9110 Section 11.6.1 https://www.rfc-editor.org/rfc/rfc9110#section-11.6.1
+export function parseWWWAuthenticate(header: string): string[] {
+    const ALPHA = 'A-Za-z';
+    const DIGIT = '0-9';
+    const tokenChar = `!#$%&'*+\\-\\.^_\`|~${DIGIT}${ALPHA}`;
+    const tchar = '[' + tokenChar + ']';
+    const OWS = '[ \\t]*';
+    const RWS = '[ \\t]+';
+    const BWS = OWS;
+    const qdtext = '[\\t\\x21\\x23-\\x5B\\x5D-\\x7E\\x80-\\xFF]';
```

Comment:
Shouldn't this include a SP character?
```suggestion
    const qdtext = '[\\t \\x21\\x23-\\x5B\\x5D-\\x7E\\x80-\\xFF]';
```

Additionally, to reduce the number of backslashes, may I suggest:
```suggestion
    const qdtext = /[\t \x21\x23-\x5B\x5D-\x7E\x80-\xFF]/.source;
```
https://www.rfc-editor.org/rfc/rfc9110#section-5.6.4

Reply:
updated

Reply:
regarding the additional backslashes, using a regex then `.source` makes eslint complains.

Reply:
The space character still appears to be missing in `qdtext`